ziti/cmd/ascode/download/output.go

```diff
@@ -0,0 +1,140 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package download
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v3"
+
+	"io"
+	"os"
+)
+
+type Output struct {
+	outputJson bool
+	outputYaml bool
+	filename   string
+	writer     *bufio.Writer
+	verbose    bool
+}
+
+func NewOutputToFile(verbose bool, outputJson bool, outputYaml bool, filename string) (*Output, error) {
+	file, err := os.Create(filename)
+	if err != nil {
+		log.WithError(err).Error("Error creating file for writing")
+		return nil, err
+	}
+	writer := bufio.NewWriter(file)
+	output, err := NewOutputToWriter(verbose, outputJson, outputYaml, writer)
+	output.filename = filename
+	return output, err
+}
+
+func NewOutputToWriter(verbose bool, outputJson bool, outputYaml bool, writer io.Writer) (*Output, error) {
+	output := Output{}
+	output.verbose = verbose
+	output.outputJson = outputJson
+	output.outputYaml = outputYaml
+	output.writer = bufio.NewWriter(writer)
+	return &output, nil
+}
+
+func (output Output) Write(data any) error {
+	var formatted []byte
+	var err error
+	if output.outputYaml {
+		if output.verbose {
+			_, _ = fmt.Fprintf(os.Stderr, "\u001B[2KFormatting as Yaml\r\n")
```

Comment:
can you make these Fprintf's a function and describe what `\u001B[2K` is doing and refactor?